TF Pro exam: New content (#2316)
hashicorp/dev-portal: src/content/certifications/exam-faqs/terraform-authoring-operations-professional.mdx

<!-- Each "## Heading Two" section in this document will be used to populate an FAQ item -->

## Exam Availability

Coming soon! 

## Prerequisites

> We strongly recommend passing the associate-level Terraform exam before taking the professional-level exam. Practitioners who have extensive experience with Terraform in a production environment, and understand the concepts covered in the associate exam, may be able to successfully pass the professional-level exam.

Reviewer: ```suggestion
We strongly recommend that you pass the associate-level Terraform exam before taking the professional-level exam. The professional-level exam is intended for practitioners who have extensive experience with Terraform in a production environment, and understand the concepts covered in the associate exam.
```

- HashiCorp Certified: Terraform Associate Certification (recommended)

> - Linux skills such as list and edit files via command terminal

Reviewer: ```suggestion
- Linux skills such as the ability to list and edit files via command terminal
```

Do we know what editors are available to the reader to edit files via the command line? Do we intended that they'll be using vim/emacs/nano/etc? Or will they have access to a full text editor such as VS Code?

| 4c | Refactor a module and use module versioning     |
| 4d | Refactor an existing configuration into modules |
| **5**  | **Configure and use Terraform providers**                                            |
| 5a | Understand Terraform's plugin-based architecture                                     |
| 5b | Configure providers, including aliasing, versioning, sourcing, and managing upgrades |
| 5c | Manage provider authentication                                                       |
| 5d | Troubleshoot provider errors                                                         |
| **6**  | **Collaborate on infrastructure as code using Terraform Cloud**                                 |
| 6a | Analyze the Terraform Cloud run workflow                                                           |
| 6b | Understand Terraform Cloud workspaces and their configuration options, including access management |
| 6c | Manage provider credentials in Terraform Cloud                                                     |
| 6d | Analyze policy as code and governance features                                                     |

## Lab-based Exam

> This performance-based exam contains labs that must be completed in a virtual environment and a shorter multiple-choice section. During the lab scenarios, exam-takers will be tested on performing real-world Terraform tasks using the command line, cloud providers' (AWS, GCP, or Azure) UI, and text editors. Exam-takers will have access to Terraform documentation, cloud provider documentation, and limited Registry access.

Reviewer: ```suggestion
This performance-based exam contains labs that must be completed in a virtual environment along with a shorter multiple-choice section. During the lab scenarios, you will be tested on performing real-world Terraform tasks using the command line, cloud providers' (AWS, GCP, or Azure) UI, and text editors. You will have access to Terraform documentation, cloud provider documentation, and limited Registry access.
```
